### DISCUSSION TITLE
Create 3 Freezes and Reboots Regularly with Firmware G.3.1

### DISCUSSION THREAD

**Discussion by scharbonneau8661** [iRobotEducation/create3_docs#215] [Questions]
Hello,

My Create 3 robot freezes and reboots regularly ever since I upgrade to Firmware 3.1.1.  The ring light turns red and the robot reboots sometime after.  This happens at least a few times a day.  When the light ring turns red, I cannot access the robot Web server.

Any help on this would be greatly appreciated.

Thank you and best regards,

Serge

**Comment by shamlian** [comment]
Could you please attach a log, if you are able to get one? If you aim your web browser at http://[ip address of your robot]/logs-raw , you should be able to get something when this load spike starts, though it may take a few tens of seconds to initiate.

Are there other robots on your network? Are you in a home network or an enterprise one? Is the robot actively in use when this happens, or does it seem random?

**Reply by scharbonneau8661** [reply]
Find the log below.

And here are the answers to your questions:

1. I only have this single robot (Alfred).
2. I'm on a home network.
3. It usually happens when the robot is on its docking station.  However, I have a python script that constantly monitor its sensors through 'ros2 topic echo' commands at all times even when it is docked and idle.

Also, I didn't have these issues with the previous version of the firmware.

Thank you for the quick reply.

[log.txt](https://github.com/iRobotEducation/create3_docs/files/9656725/log.txt)

**Reply by shamlian** [reply]
Things look more or less normal in this log, to me; I assume this was taken just now and not while the red light event is occurring. If possible, it would be good to get a log while (or after) the light ring turns red. When you say the robot "reboots," I assume you mean that the application restarts, and the robot does not hard power off?

**Reply by scharbonneau8661** [reply]
Yes, the robot restarts, not reboots.  I know because it plays its chime each time.  I realize it's happening even when I am not in the home office where the robot is located by default.  I will send you the log at the next occurrence of a restart.  Also, I will not run my python script to see if it restarts even without this running.  The script was also running when I had the previous firmware version installed on the robot and it did not restart at random like it is doing now.  This happens maybe a few times a day.  Like 2-3 times.  Not more.

**Reply by shamlian** [reply]
I'm sorry that you've had this trouble, and we would love to understand why the problem is happening; if you are able to get a log after the application restarts, hopefully it would help us to understand the cause of the problem so that we can fix it. Feel free to keep your Python script running if you feel it is the trigger (and if you don't mind, we'd love to see the Python script, too). I do like the idea of running the experiment without the script running to see if the event doesn't happen, as well, but I'd say when debugging it would be best to only change one thing at a time.

**Reply by scharbonneau8661** [reply]
No worries.  We'll figure it out.  The Create 3 rocks.  I've been dreaming about have a robotic platform like this one for 10 years now, 10 years ago I was hacking my Roomba 550 with a Rootooth device and talking to it via bluetooth.

**Reply by shamlian** [reply]
Any updates on this? Hoping your robot is behaving itself.

**Reply by scharbonneau8661** [reply]
My robot is behaving most of the time, but I still have the problem identified above on occasions.  I only have this problem when my Python script runs.  When the robot is completely idle, I don't get this problem.  The script executes a 'ros2 topic echo' command for each topic for 5 seconds (15 seconds for the battery_state topic).  It launches the command, waits 5 seconds (or 15), stops the command and collects the results.  It does it for each topic one at a time continuously.  I notice that I get the following error message running a 'ros2 topic echo' command when the light ring turns red prior to the robot restarting:

2022-10-12 06:55:41.232 [RTPS_TRANSPORT_SHM Error] Failed to create segment 6dfbcded97e123a5: boost::interprocess_exception::library_error -> Function compute_per_allocation_extra_size
2022-10-12 06:55:41.232 [RTPS_MSG_OUT Error] boost::interprocess_exception::library_error -> Function init

I remember reading there is a problem when RMW_IMPLEMENTATION is set to rwm_fastrtsp_cpp which is exactly what I am running.  Can this be causing the issue I'm experiencing?

**Reply by justinIRBT** [reply]
That does look like the RMW layer complaining, what are the memory/CPU printouts in the log around the time this error occurs?  They look like:
```
Sep 27 14:06:41 Alfred user.notice create-platform: [INFO] [1664287601.864329542] [alfred.system_health]: CPU usage: max 72 [%] mean 58 [%] RAM usage: 36/59 [MB]
```
I am not personally aware of an issue with the usage model you are describing.
Have you tried using cyclone to as the RMW to see if the same error occurs?
